src/kokkos_abstraction.hpp

```diff
@@ -635,6 +638,42 @@ struct SpaceInstance<Kokkos::Cuda> {
 };
 #endif
 
+// Design from "Runtime Polymorphism in Kokkos Applications", SAND2019-0279PE
+template <typename MS = DevMemSpace>
+struct DeviceDeleter {
+  template <typename T>
+  void operator()(T *ptr) {
+    Kokkos::kokkos_free<MS>(ptr);
+  }
+};
+
+template <typename T, typename ES = DevExecSpace, typename MS = DevMemSpace>
+std::unique_ptr<T, DeviceDeleter<MS>> DeviceAllocate() {
+  PARTHENON_REQUIRE(std::is_trivially_destructible<T>::value,
+                    "DeviceAllocate only supports trivially destructible classes!");
```

Comment:
Sorry, should have caught this earlier.
```suggestion
  static_assert(std::is_trivially_destructible<T>::value,
                    "DeviceAllocate only supports trivially destructible classes!");
```

Reply:
thanks!